src/main/groovy/org/mockito/release/internal/gradle/PublicationsComparatorPlugin.java

```diff
@@ -75,23 +118,26 @@ public void execute(final PublicationsComparatorTask t) {
                     @Override
                     public void run() {
                         t.setProjectGroup(project.getGroup().toString());
-                        /*
-                        TODO ww it is more convenient for Gradle users and for us, when the task has simple inputs (Strings, Files)
-                        If the task API has getters/setters on types like DefaultArtifactUrlResolver
-                        it is harder to make it incremental, it also makes the public API bigger because you need to expose and document those types
-                        Why not have the url get calculated right here, with some conditional check:
-
-                        if (t.getPreviousJarUrl() == null) {
-                          t.setPreviousJarUrl(...)
-                        }
-
-                        */
-                        DefaultArtifactUrlResolver artifactUrlResolver =
-                                new DefaultArtifactUrlResolverFactory().getDefaultResolver(project, sourcesJar.getBaseName(), conf.getPreviousReleaseVersion());
-                        t.setDefaultArtifactUrlResolver(artifactUrlResolver);
                     }
                 });
             }
         });
     }
+
+    private String getBasePreviousVersionArtifactPath(Project project, ReleaseConfiguration conf, Jar sourcesJar) {
+        return project.getBuildDir().getAbsolutePath() + PREVIOUS_RELEASE_ARTIFACTS_DIR
+                + File.separator + sourcesJar.getBaseName() + "-" + conf.getPreviousReleaseVersion();
+    }
+
+    private String getDefaultIfNull(String url, String variableName, String extension, DefaultArtifactUrlResolver defaultArtifactUrlResolver) {
+        if(url == null){
+            if(defaultArtifactUrlResolver == null){
+                return null;
+            }
+            String defaultUrl = defaultArtifactUrlResolver.getDefaultUrl(extension);
+            LOG.lifecycle("Variable {} not set. Setting it to default value - {}", variableName, defaultUrl);
```

Comment:
Is this 'lifecycle' message useful? From looking at it, it seems like a 'info' level message. Also users will not know what the 'variable' is (they would wonder if this is an env variable? is it a property on a task?).